dockstore-common/src/main/java/io/dockstore/common/ValidationConstants.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.common;
+
+import java.util.regex.Pattern;
+
+/**
+ * This file describes the constraints on various Dockstore inputs.
+ */
+public final class ValidationConstants {
+
+    public static final String ENTRY_NAME_REGEX = "[a-zA-Z0-9]+([-_][a-zA-Z0-9]+)*+";
+    public static final Pattern ENTRY_NAME_PATTERN = Pattern.compile(ENTRY_NAME_REGEX);
+    public static final String ENTRY_NAME_REGEX_MESSAGE = "must contain only letters, numbers, and internal hyphens and underscores";
```

Comment:
maybe this should say alphanumeric. Presumably letters would be less clear that we don't support accents, greek letters, etc.

Reply:
IMHO, the two are synonymous, and "alphanumeric" sounded more stilted to me, so I went with "letters" and "numbers".

Reply:
This is not my understanding, alphanumeric implies Arabic numerals and Latin characters https://en.wikipedia.org/wiki/Alphanumericals 

Letters are just any language that uses letters https://en.wikipedia.org/wiki/Letter_(alphabet)

That said, if it isn't common knowledge, maybe just spell out what kind of letters you're using.

Reply:
https://imgflip.com/i/6yjwta

Reply:
![6yjxoc](https://user-images.githubusercontent.com/1730679/198401608-0327469d-a878-4462-ad02-02ee749e1e41.jpg)

Reply:
Why are we limiting entry names to characters within the ASCII range? This could bite us some day in non-English speaking countries. This is probably beyond the scope of this PR and the PR may be existing behavior (we have the same issue with GitHub branch names), but we should start thinking about support beyond ASCII at some point.

Reply:
Probably similar to some of the concerns about sourcefile names.

Reply:
I replaced the notice text with a passage that I extracted from our docs.